R/trans-compose.R

```diff
@@ -0,0 +1,51 @@
+#' Compose two or more transformations together
+#'
+#' This transformer provides a general mechanism for composing two or more
+#' transformers together. The most important use case is to combine reverse
+#' with other transformations.
+#'
+#' @param ... One or more transformers, either specified with string or
+#'   as individual transformer objects.
+#' @export
+#' @examples
+#' demo_continuous(10^c(-2:4), trans = "log10", labels = label_log())
+#' demo_continuous(10^c(-2:4), trans = c("log10", "reverse"), labels = label_log())
+compose_trans <- function(...) {
+  trans_list <- lapply(list2(...), as.trans)
+  if (length(trans_list) == 0) {
+    abort("Must include at least 1 transformer to compose")
+  }
+
+  # Resolve domains
+  suppressWarnings(
+    domain <- compose_fwd(trans_list[[1]]$domain, trans_list[-1])
+  )
+  if (any(is.na(domain))) {
+    abort("Sequence of transformations yields invalid domain")
+  }
+  domain <- range(domain)
+
+  names <- vapply(trans_list, "[[", "name", FUN.VALUE = character(1))
+
+  trans_new(
+    paste0("composition(", paste0(names, collapse = ","), ")"),
```

Comment:
Apologies for nitpicking, but this might print a little bit nicer with a space behind the comma, e.g.:
`paste0("composition(", paste0(names, collapse = ", "), ")")`

Reply:
I get where you're coming from, but I wanted this to look like a single "name", i.e. no spaces.